packages/teams-js/src/public/app.ts

```diff
@@ -534,11 +534,16 @@ export namespace app {
    * @returns Promise that will be fulfilled when initialization has completed, or rejected if the initialization fails or times out
    */
   export function initialize(validMessageOrigins?: string[]): Promise<void> {
-    return runWithTimeout(
-      () => initializeHelper(validMessageOrigins),
-      initializationTimeoutInMs,
-      new Error('SDK initialization timed out.'),
-    );
+    if (typeof window !== 'undefined') {
```

Comment:
>if (typeof window !== 'undefined') [](http://example.com/codeflow?start=4&length=34)

Consider extracting this check into a more semantically meaningful function you can call, e.g.:
```
if (!inServerSideRenderingEnvironment()) {
...
} else {
...
}